From 31a1ec780aaba33c1008b8dc7bca7da88bd297ff Mon Sep 17 00:00:00 2001
From: Jocelyn Turcotte <jocelyn.turcotte@digia.com>
Date: Tue, 28 Jan 2014 13:07:59 +0100
Subject: [PATCH] Don't pollute .prl & .pc files with private dependencies

Also change the creating_module check in needToLink to make sure that
only the main module will try to link directly with the WebCore and
JavaScriptCore static libraries.

Task-number: QTBUG-35774
Initial-patch-by: Oswald Buddenhagen <oswald.buddenhagen@digia.com>
Change-Id: Ib5992e112803679837249400e98cb8c22636096d
---
 Source/widgetsapi.pri                         | 1 +
 Tools/qmake/mkspecs/features/default_post.prf | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Source/widgetsapi.pri b/Source/widgetsapi.pri
index 7bdc0e1..335e371 100644
--- a/Source/widgetsapi.pri
+++ b/Source/widgetsapi.pri
@@ -49,6 +49,7 @@ use?(3D_GRAPHICS): WEBKIT += ANGLE
 WEBKIT += javascriptcore wtf webcore
 
 MODULE = webkitwidgets
+CONFIG += creating_module
 
 # We want the QtWebKit API forwarding includes to live in the root build dir.
 MODULE_BASE_DIR = $$_PRO_FILE_PWD_
diff --git a/Tools/qmake/mkspecs/features/default_post.prf b/Tools/qmake/mkspecs/features/default_post.prf
index 6fae22f..548005d 100644
--- a/Tools/qmake/mkspecs/features/default_post.prf
+++ b/Tools/qmake/mkspecs/features/default_post.prf
@@ -171,8 +171,8 @@ contains(QT, webkit)|contains(QT, webkitwidgets) {
 }
 
 defineTest(needToLink) {
-    # The main dynamic library always needs to link in the intermediate libraries
-    creating_module: return(true)
+    # The main module dynamic library always needs to link in the intermediate libraries
+    isEqual(MODULE, webkit): return(true)
 
     force_static_libs_as_shared {
         # So do the individual libraries if they are dynamic
-- 
1.9.0

